modules/drivers/bigquery-cloud-sdk/src/metabase/driver/bigquery_cloud_sdk/query_processor.clj

```diff
@@ -431,6 +431,24 @@
     [_ _ expr]
     (with-temporal-type (hsql/call bigquery-fn expr) :timestamp)))
 
+(defmethod sql.qp/->honeysql [:bigquery-cloud-sdk :convert-timezone]
+  [driver [_ arg target-timezone source-timezone]]
+  (let [timestamp    (partial hsql/call :timestamp)
+        datetime     (partial hsql/call :datetime)
+        clause       (sql.qp/->honeysql driver arg)
+        timestamptz? (hx/is-of-type? clause "timestamp")]
+    (when (and timestamptz? source-timezone)
+      (throw (ex-info (tru "`timestamp` columns shouldn''t have a `source timezone`")
+                      {:type            qp.error-type/invalid-query
+                       :target-timezone target-timezone
+                       :source-timezone source-timezone})))
+    (let [source-timezone (or source-timezone (qp.timezone/results-timezone-id))]
+      (cond-> clause
+        (and (not timestamptz?) source-timezone)
+        (timestamp source-timezone)
```

Comment:
Shouldn't convert-timezone always return a datetime, since it needs to be without a time zone?

Reply:
I'm not following, isn't that how I'm implementing here with the datetime call at the end?